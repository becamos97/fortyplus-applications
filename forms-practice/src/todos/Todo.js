import React from "react";

export default function Todo({ task, removeTodo }) {
  return (
    <li>
      <span>{task}</span>{" "}
      <button aria-label="remove-todo" onClick={removeTodo}>X</button>
    </li>
  );
}