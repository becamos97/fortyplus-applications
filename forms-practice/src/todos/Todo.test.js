import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Todo from "./Todo";

it("renders without crashing", () => {
  render(<Todo task="Test" removeTodo={() => {}} />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Todo task="Test" removeTodo={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

it("calls remove on click", () => {
  const mock = jest.fn();
  render(<Todo task="Delete me" removeTodo={mock} />);
  fireEvent.click(screen.getByLabelText(/remove-todo/i));
  expect(mock).toHaveBeenCalled();
});