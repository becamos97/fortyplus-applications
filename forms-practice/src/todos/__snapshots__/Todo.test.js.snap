// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li>
    <span>
      Test
    </span>
     
    <button
      aria-label="remove-todo"
    >
      X
    </button>
  </li>
</DocumentFragment>
`;
